File: src/main/java/net/hectus/neobb/turn/person_game/warp/PTVoidWarp.java
package net.hectus.neobb.turn.person_game.warp;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.structure.PlacedStructure;
import org.bukkit.World;
import org.bukkit.potion.PotionEffectType;

public class PTVoidWarp extends PWarpTurn {
    public PTVoidWarp(World world) { super(world, "void"); }
    public PTVoidWarp(PlacedStructure data, World world, NeoPlayer player) { super(data, world, player, "void"); }

    @Override
    public void apply() {
        super.apply();
        player.game.players().forEach(p -> p.inventory.removeRandom());
        new Buff.Effect(Buff.BuffTarget.OPPONENTS, PotionEffectType.BLINDNESS).apply(player);
        player.game.turnScheduler.runTaskLater("blindness", () -> // Blindness only lasts for 3 turns.
                player.opponents(true).forEach(p -> p.player.removePotionEffect(PotionEffectType.BLINDNESS)), 3);
    }

    @Override
    public int chance() {
        return 20;
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/person_game/warp/PTVoidWarp.kt
package net.hectus.neobb.turn.person_game.warp

import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.Effect
import net.hectus.neobb.game.util.ScheduleID
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.structure.PlacedStructure
import org.bukkit.potion.PotionEffectType

class PTVoidWarp(data: PlacedStructure?, player: NeoPlayer?) : PWarpTurn(data, "void", player) {
    override val chance: Double = 20.0

    override fun apply() {
        player!!.game.players.forEach { it.inventory.removeRandom() }
        Effect(PotionEffectType.BLINDNESS, target = Buff.BuffTarget.OPPONENTS).apply(player)

        player.game.turnScheduler.runTaskLater(ScheduleID.BLINDNESS, 3) {
            player.opponents().forEach { it.player.removePotionEffect(PotionEffectType.BLINDNESS) }
        }
    }
}
